active-rfcs/0001-issue-helper.md

```diff
@@ -0,0 +1,105 @@
+- Implementation PR:
+
+# Summary
+
+We'll use GitHub Issue [contact_links](https://docs.github.com/en/github/building-a-strong-community/configuring-issue-templates-for-your-repository#configuring-the-template-chooser) to create issues, and in contact_links we'll put an external web page called issue-helper to help us create issues in a standard format.
+
+# Motivation
+
+In the past, TiDB's code repository used GitHub Issues to create bug reports, but because GitHub Issues use markdown to organize and write the report content. This form is simple but not structured, and not easy to use, so information transmission is inefficient.
+
+Also, as our bug reports require more and more information, the issue templates become more and more complex, making it difficult for users or contributors to fill out the templates.
+
+So we can use GitHub Issue's contact_links to develop a web page to help and guide people through the process of filling out a bug report. This will help us collect bug reports in a better and more standard way and make it easier for users and contributors to fill them out.
+
+# Detailed design
+
+## Introducing GitHub Issue contact_links
+
+GitHub Issues provides a way to configure external links. We can use the following configuration to add an external contact link.
+
+```yaml
+// .github/ISSUE_TEMPLATE/config.yml
+blank_issues_enabled: false
+contact_links:
+  - name: Create new issue
+    url: https://prow.tidb.io/tichi/new-issue
+    about: Please use the following link to create a new issue.
+```
+
+The effect of using this feature can be seen in the [vuejs repository](https://github.com/vuejs/vue/issues/new/choose).
+
+## External contact links combined with GitHub Issues
+
+Once we enable external links, we can direct the user or contributor to a web page that we've developed, where we'll use a standard form for the user or contributor to fill out the report, and then we'll use the [GitHub Issue creation API](https://docs.github.com/en/github/managing-your-work-on-github/about-automation-for-issues-and-pull-requests-with-query-parameters#supported-query-parameters) to resubmit the form to GitHub and create the Issue.
+
+The API looks like this: https://github.com/pingcap/tidb/issues/new?title=example&body=FormContent
+
+## Issue Helper
+
+The main issues we need to focus on in TiDB are divided into two categories, one is [Bug Report](https://github.com/pingcap/tidb/blob/master/.github/ISSUE_TEMPLATE/bug-report.md) and the other is [New Feature Request](https://github.com/pingcap/tidb/blob/master/.github/ISSUE_TEMPLATE/feature-request.md).
+
+So our Issue Helper focuses on these two types of reports and develops forms for them and then fills them back into the GitHub Issue.
+
+### Bug Report Template
+
+```md
+### Steps to reproduce
+
+### What is expected?
+
+### What is actually happening?
+
+### Any additional comments? (optional)
+
+| Environment   | Info   |
+| ------------- | ------ |
+| TiDB Version  | v4.0.1 |
```

Comment:
I found that in the current repository, the version information filled in the bug type issue will be very specific, not only the version number but also the commit and other information. Such as: https://github.com/pingcap/tidb/issues/23036

Reply:
In the case of bug reproduction purposes, it is necessary to keep the specific version number. For a non-release compiled binary of the system which is reported to have a bug, the commit should be very useful. First to reproduce, then next probably it will be used for root cause location (e.g. introduced by which previous commit). I would say it is important to record such a version/commit in the helper.

Reply:
After investigating the user's usage and feedback, the requirements for the filled-in version number are roughly as follows

1. Support to fill in the version number of pd/tidb/tikv separately
2. Support select release version
3. Support filling in non-released versions, such as commit hash/tag
4. Consider supporting multiple versions

Reply:
> 4\. Consider supporting multiple versions

Should this be the main version like `5.x and 4.x` or minor release `5.0.1 and 5.0.2`? I'm worried there will be too many possibilities...